uwblueprint/planet-read
# Add translation progress bar

## Notion ticket link
[Add translation progress state](https://www.notion.so/uwblueprintexecs/Add-translation-progress-state-d384e35a5f3e49bf89d3ed64d7e52573)


## Implementation description
* Added progress bar to TranslationPage in frontend
* Counting and returning number of translated lines on `storyTranslationById` query (added `numTranslatedLines` to `StoryTranslationResponseDTO`)
* Will be building implementation in frontend to reflect/update percentage progress bar in frontend after PR #39 is merged


## Steps to test
1. Seed database with data:
```
$ INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('PlanetAdmin', 'Read', <auth id>, <role>, '{"HINDI":3}');
$ INSERT INTO stories (title, description, youtube_link, level, translated_languages) VALUES ('A Book for Puchku', 'Puchku has run out of books to read. Then she discovers more books in the top shelf of the library bookcase. But Puchku is small, and the bookcase tall. How will she ever get to her beloved books?', 'https://www.youtube.com/watch?v=PKxMjO4HPvE', 3, '{}');
$ INSERT INTO story_contents (story_id, line_index, content) VALUES (<id of story you just inserted>,1,'first line'), (<id of story you just inserted>,2,'second line'), (<id of story you just inserted>,3,'third line'), (<id of story you just inserted>,4,'fourth line');
$ INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (<id of story>,'HINDI', 'TRANSLATE',1);
$ INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES (<id of story_translation>, 1, 'translation!');
$ INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES (<id of story_translation>, 2, '');
$ INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES (<id of story_translation>, 3, '');
```
2. Check that query for storyTranslationById returns correct number of translated lines (i.e. in this case it should return 1):
```
{
  storyTranslationById(id: 1) {
    id
    numTranslatedLines
  }
}
```
3. Verify the CSS for the progress bar when you navigate to `http://localhost:3000/translation/{storyId}/{storyTranslationId}` (currently hardcoded at 25%)
![image](https://user-images.githubusercontent.com/32009013/125223802-5ee77480-e29a-11eb-93a6-b85f11e11d9a.png)

## What should reviewers focus on?
* `backend/python/app/services/implementations/story_service.py`
* `frontend/src/components/translation/TranslationProgressBar.tsx`


## Checklist
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."`
